gnomad_hail/utils/variant_qc.py

```diff
@@ -263,80 +275,481 @@ def quantiles_to_bin_boundaries(quantiles: List[int]) -> Dict:
                     global_bin_indices=hl.tarray(hl.tint32),
                     merged_bins=hl.tdict(hl.tint32, hl.tint32)
-                ) for rank_id in rank_expr
+                ) for bin_id in bin_expr
             })
         )
     )
 
-    # Annotate the rank as the index in the unique boundaries array
-    ht = ht.annotate(
+    # Annotate the bin as the index in the unique boundaries array
+    bin_ht = bin_ht.annotate(
         **{
-            rank_id: hl.or_missing(
-                ht[f'_filter_{rank_id}'],
-                hl.binary_search(ht.rank_stats[rank_id].bin_boundaries, ht._score),
-            ) for rank_id in rank_expr
+            bin_id: hl.or_missing(
+                bin_ht[f'_filter_{bin_id}'],
+                hl.binary_search(bin_ht.bin_stats[bin_id].bin_boundaries, bin_ht._score),
+            ) for bin_id in bin_expr
         }
     )
 
-    # Convert the rank to global rank by expanding merged bins, that is:
+    # Convert the bin to global bin by expanding merged bins, that is:
     # If a value falls in a bin that needs expansion, assign it randomly to one of the expanded bins
-    # Otherwise, simply modify the rank bin to its global index (with expanded bins that is)
-    ht = ht.select(
+    # Otherwise, simply modify the bin to its global index (with expanded bins that is)
+    bin_ht = bin_ht.select(
         **{
-            rank_id: hl.cond(
-                ht.rank_stats[rank_id].merged_bins.contains(ht[rank_id]),
-                ht[rank_id] + hl.int(hl.rand_unif(0, ht.rank_stats[rank_id].merged_bins[ht[rank_id]] + 1)),
-                ht.rank_stats[rank_id].global_bin_indices[ht[rank_id]]
+            bin_id: hl.cond(
+                bin_ht.bin_stats[bin_id].merged_bins.contains(bin_ht[bin_id]),
+                bin_ht[bin_id] + hl.int(hl.rand_unif(0, bin_ht.bin_stats[bin_id].merged_bins[bin_ht[bin_id]] + 1)),
+                bin_ht.bin_stats[bin_id].global_bin_indices[bin_ht[bin_id]]
             )
-            for rank_id in rank_expr
+            for bin_id in bin_expr
         }
     )
 
     if desc:
-        ht = ht.annotate(
-            **{rank_id: n_bins - ht[rank_id] for rank_id in rank_expr}
+        bin_ht = bin_ht.annotate(
+            **{bin_id: n_bins - bin_ht[bin_id] for bin_id in bin_expr}
         )
 
+    # Because SNV and indel rows are mutually exclusive, re-combine them into a single bin.
+    # Update the global bin_stats struct to reflect the change in bin names in the table
+    if stratify_snv_indel:
+        bin_expr_no_snv = {bin_id.rsplit("_", 1)[0] for bin_id in bin_ht.bin_stats}
+        bin_ht = bin_ht.annotate_globals(
+            bin_stats=hl.struct(
+                **{
+                    bin_id: hl.struct(
+                        **{snv: bin_ht.bin_stats[f"{bin_id}_{snv}"] for snv in ["snv", "indel"]}
+                    )
+                    for bin_id in bin_expr_no_snv
+                }
+            )
```

Comment:
I had to add this because a later step in the pipeline gets screwed up because it uses bin_stats